### PR TITLE
Feature/Fio address registration

### PR DESCRIPTION
**Pull request by andreyvEze**

### CODE REVIEW

File: src/actions/WalletOptionsActions.js [hunk 1]
```diff
@@ -53,8 +54,31 @@ export const walletRowOption = (walletId: string, option: string, archived: bool
...
-      return (dispatch: Dispatch) => {
-        dispatch(showDeleteWalletModal(walletId))
+      return async (dispatch: Dispatch, getState: GetState) => {
+        const state = getState()
+        const wallet = WALLET_SELECTORS.getWallet(state, walletId)
+        const type = wallet.type.replace('wallet:', '')
+        let fioAddress = ''
+
+        if (type === 'fio') {
+          const engine = getState().core.account.currencyWallets[walletId]
```
Comment (paullinator) [inline src/actions/WalletOptionsActions.js]: use `state`

File: src/components/scenes/FioAddressConfirmScene.js [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+      try {
+        const obj = await wallet.otherMethods.fioAction('getFee', { endPoint: 'register_fio_address', fioAddress: '' })
+        if (obj) {
+          if (obj.fee) {
+            const displayFee = obj.fee / Constants.BILLION
+            this.setState({
+              fee: obj.fee,
+              displayFee
+            })
+            return displayFee
```
Comment (paullinator) [inline src/components/scenes/FioAddressConfirmScene.js]: why return?

File: src/components/scenes/FioAddressConfirmScene.js [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+      try {
+        const obj = await wallet.otherMethods.fioAction('getFioBalance', {})
+
+        if (obj) {
+          if (obj.balance || obj.balance === 0) {
+            const newBalance = obj.balance / Constants.BILLION
+            this.setState({
+              balance: newBalance
+            })
+            return newBalance
```
Comment (paullinator) [inline src/components/scenes/FioAddressConfirmScene.js]: why return?

File: src/locales/en_US.js [hunk 4]
```diff
@@ -585,7 +593,41 @@ const strings = {
...
+  fio_address_register_first_free: 'Register First Free FIO Address',
+  fio_address_register_custom_domain: 'Register custom domain',
+  fio_address_domain_already_own: 'Use domain I already own',
+  fio_address_register_screen_label: 'Create a unique FIO Address',
+  fio_address_register_form_field_label: 'FIO Address',
+  fio_address_register_screen_available: 'Available',
+  fio_address_register_screen_not_available: 'Not Available',
+  fio_address_register_screen_invalid: 'Invalid name',
+  fio_address_register_default_fio_wallet_name: 'FIO Wallet',
+  fio_address_register_screen_button_next: 'Next',
```
Comment (paullinator) [inline src/locales/en_US.js]: dupe

File: src/util/utils.js [hunk 5]
```diff
@@ -628,3 +630,31 @@ export const getYesterdayDateRoundDownHour = () => {
...
+export const getMonthName = (month: number, short: boolean = false) => {
+  return short ? MONTH_NAMES_SHORT[month] : MONTH_NAMES[month]
+}
+
+export const formatExpDate = (expiration: Date, short: boolean = false) => {
+  const expirationDate = new Date(expiration)
+  return `${getMonthName(expirationDate.getMonth())}  ${expirationDate.getDate()}, ${expirationDate.getFullYear()}`
+}
+
+export const networkAlert = (): boolean => {
```
Comment (paullinator) [inline src/util/utils.js]: Do not use system alert. Use showError instead.

File: src/util/utils.js [hunk 6]
```diff
@@ -628,3 +630,31 @@ export const getYesterdayDateRoundDownHour = () => {
...
+  const dec = number % 10
+
+  if (dec) {
+    return dec.toString().length > 2 ? number.toString() : number.toFixed(defaultAmount)
+  }
+
+  return number.toFixed(defaultAmount)
+}
+
+export const validateFioAddress = (fioAddress: string): boolean => {
```
Comment (paullinator) [inline src/util/utils.js]: This should be provided by the currency plugin and exported via otherMethods

File: yarn.lock [hunk 7]
```diff
@@ -9816,12 +9821,51 @@ react-native-mail@3.0.7, react-native-mail@^3.0.5:
   resolved "https://registry.yarnpkg.com/react-native-mail/-/react-native-mail-3.0.7.tgz#07e6aca43f6f1f85f3e58e96e7fcd16b31a927b8"
   integrity sha512-FHe4kKJKAGuCPPWcDwaRSjdUhEE4IWIsVtTZ05dgD/MgOm00isYmM20zuJHla7oiHVQ2HmzGji76g6IuhIrCjw==
 
+react-native-material-buttons@^0.5.0:
```
Comment (paullinator) [inline yarn.lock]: This does not appear to be used

File: yarn.lock [hunk 8]
```diff
@@ -9816,12 +9821,51 @@ react-native-mail@3.0.7, react-native-mail@^3.0.5:
...
+    react-native-material-ripple "^0.8.0"
+    react-native-material-textfield "^0.12.0"
+
 "react-native-material-kit@git://github.com/xinthink/react-native-material-kit.git#da1c292e":
   version "0.5.1"
   resolved "git://github.com/xinthink/react-native-material-kit.git#da1c292e2a6036b570381edb7a3752f254daf2b3"
   dependencies:
     ramda "^0.25.0"
 
+react-native-material-ripple@^0.7.0:
```
Comment (paullinator) [inline yarn.lock]: This does not appear to be used

File: src/components/materialWrappers/Select.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import React, { Component } from 'react'
+import { Dropdown } from 'react-native-material-dropdown'
```
Comment (paullinator) [inline src/components/materialWrappers/Select.js]: Screenshot of this component? We probably have some other UI that can be used instead.

File: src/util/utils.js [hunk 10]
```diff
@@ -628,3 +630,31 @@ export const getYesterdayDateRoundDownHour = () => {
...
+export const formatExpDate = (expiration: Date, short: boolean = false) => {
+  const expirationDate = new Date(expiration)
+  return `${getMonthName(expirationDate.getMonth())}  ${expirationDate.getDate()}, ${expirationDate.getFullYear()}`
+}
+
+export const networkAlert = (): boolean => {
+  return Alert.alert(s.strings.network_alert_title, s.strings.network_alert_text)
+}
+
+export const getFeeDisplayed = (number: number) => {
```
Comment (paullinator) [inline src/util/utils.js]: Why not just always use `toFixed(2)`?

File: src/util/utils.js [hunk 11]
```diff
@@ -628,3 +630,31 @@ export const getYesterdayDateRoundDownHour = () => {
   const yesterday = date.setDate(date.getDate() - 1)
   return new Date(yesterday).toISOString()
 }
+
+export const getMonthName = (month: number, short: boolean = false) => {
```
Comment (paullinator) [inline src/util/utils.js]: This does not return localized formats or month names. Use IntlDateTimeFormat() instead. This should be placed in Intl.js

File: src/components/Main.ui.js [hunk 12]
```diff
@@ -800,6 +884,14 @@ export default class Main extends Component<Props> {
       handlePluginBack()
       return true
     }
+    if (this.isCurrentScene(Constants.FIO_ADDRESS_DETAILS)) {
```
Comment (paullinator) [inline src/components/Main.ui.js]: Is this really needed?

File: src/components/scenes/FioAddressListScene.js [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+import s from '../../locales/strings.js'
+import { Button } from '../../modules/UI/components/ControlPanel/Component/Button/Button.ui'
+import T from '../../modules/UI/components/FormattedText/index'
+import Gradient from '../../modules/UI/components/Gradient/Gradient.ui'
+import SafeAreaView from '../../modules/UI/components/SafeAreaView/index'
+import styles from '../../styles/scenes/FioAddressListStyle'
+import type { GoToSceneProps } from '../../types/types'
+
+export type State = {
+  walletAddresses: Array<Object>
```
Comment (paullinator) [inline src/components/scenes/FioAddressListScene.js]: Add flow type for Object

File: src/components/scenes/FioAddressRegisterScene.js [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+  isAvailable: boolean | null,
+  replaceRegex: RegExp
+}
+
+export type StateProps = {
+  fioWallets: EdgeCurrencyWallet[]
+}
+
+export type DispatchProps = {
+  changeFioAddressName(fioAddressName: string): any,
```
Comment (paullinator) [inline src/components/scenes/FioAddressRegisterScene.js]: Return type should be void

File: src/components/scenes/FioAddressRegisterScene.js [hunk 15]
```diff
@@ -0,0 +1,200 @@
...
+    const { fioWallets } = this.props
+    if (fioWallets.length === 0) {
+      this.wallet = await this.createFioWallet()
+    } else {
+      this.wallet = fioWallets[0]
+    }
+    this.getRegexFromDomain(FIO_DOMAIN_DEFAULT)
+  }
+
+  createFioWallet = async () => {
```
Comment (paullinator) [inline src/components/scenes/FioAddressRegisterScene.js]: Add return type. Many other methods should also have return type added.

File: src/modules/FioAddress/action.js [hunk 16]
```diff
@@ -0,0 +1,17 @@
+// @flow
+import type { EdgeCurrencyWallet } from 'edge-core-js'
+
+export const changeFioAddressName = (fioAddressName: string) => ({
```
Comment (paullinator) [inline src/modules/FioAddress/action.js]: Do not use action creators. Instead directly dispatch the object and create a flow type for the action object.

File: src/components/scenes/FioAddressRegisterScene.js [hunk 17]
```diff
@@ -0,0 +1,200 @@
...
+  state = {
+    fioAddress: '',
+    isValid: false,
+    touched: false,
+    isAvailable: false,
+    replaceRegex: new RegExp(`${FIO_DOMAIN_DEFAULT}`)
+  }
+
+  wallet: EdgeCurrencyWallet | null
+  bcValidationTimer: any
```
Comment (paullinator) [inline src/components/scenes/FioAddressRegisterScene.js]: Type should be TimeoutID.

File: src/actions/FioActions.js [hunk 18]
```diff
@@ -4,10 +4,11 @@ import { type EdgeAccount } from 'edge-core-js/types'
 import { Linking } from 'react-native'
 
 import { showActivity, showError } from '../components/services/AirshipInstance.js'
+import { FIO_WALLET_TYPE } from '../constants/WalletAndCurrencyConstants'
 import s from '../locales/strings.js'
 import { type Dispatch, type GetState } from '../types/reduxTypes.js'
 
-export function registerFioAddress () {
+export function registerFirstFioAddress () {
```
Comment (paullinator) [inline src/actions/FioActions.js]: Appears unused.

File: src/connectors/scenes/FioAddressConfirmConnector.js [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+  const out: StateProps = {
+    fioAddressName: fioAddress.fioAddressName,
+    account,
+    fioWallets
+  }
+  return out
+}
+
+const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
+  createCurrencyWallet: (walletName: string, walletType: string) => dispatch(createCurrencyWallet(walletName, walletType, 'iso:USD', false, false)),
```
Comment (paullinator) [inline src/connectors/scenes/FioAddressConfirmConnector.js]: Should not hardcode `iso:USD`. Look up the fiat currency of the account to determine which fiat currency to use for the wallet. Even better, let the user choose.

File: src/modules/UI/selectors.js [hunk 20]
```diff
@@ -15,6 +16,19 @@ export const getWallets = (state: State) => {
   return wallets
 }
 
+export const getFioWallets = (state: State) => {
+  const { activeWalletIds = [] } = state.core.account
+  const fioWallets = []
```
Comment (paullinator) [inline src/modules/UI/selectors.js]: Returning a statically declared object will force a rerender for any mapStateToProps that use this selector.

File: src/connectors/scenes/FioAddressDetailsConnector.js [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+import type { State } from '../../types/reduxTypes'
+import { formatExpDate } from '../../util/utils'
+
+const mapStateToProps = (state: State) => {
+  const { fioAddress } = state.ui.scenes
+
+  const out: StateProps = {
+    fioAddressName: fioAddress.fioAddressName,
+    expiration: fioAddress.expiration,
+    expirationFormatted: formatExpDate(fioAddress.expiration),
```
Comment (paullinator) [inline src/connectors/scenes/FioAddressDetailsConnector.js]: expirationFormatted will always be a new string and will force a rerender on every redux change. Move the call to formatExpDate to inside the component.

File: src/components/scenes/FioAddressConfirmScene.js [hunk 22]
```diff
@@ -0,0 +1,319 @@
...
+        const fioNames = await wallet.otherMethods.fioAction('getFioNames', { fioPublicKey: receiveAddress.publicAddress })
+
+        let name = ''
+        if (fioNames) {
+          if (fioNames.fio_addresses) {
+            name = fioNames.fio_addresses.find(item => item.fio_address === fioAddressName)
+            if (name) {
+              if (name.expiration) {
+                changeConfirmSelectedWallet(selectedWallet, new Date(name.expiration), 300000)
+                setTimeout(() => Actions[Constants.FIO_ADDRESS_DETAILS]())
```
Comment (paullinator) [inline src/components/scenes/FioAddressConfirmScene.js]: Use requestAnimationFrame instead

File: src/components/scenes/FioAddressConfirmScene.js [hunk 23]
```diff
@@ -0,0 +1,319 @@
...
+              }
+            }
+          }
+        }
+      } catch (e) {
+        //
+      }
+    } else {
+      changeConfirmSelectedWallet(selectedWallet, new Date(), 300000)
+      setTimeout(() => Actions[Constants.FIO_ADDRESS_DETAILS]())
```
Comment (paullinator) [inline src/components/scenes/FioAddressConfirmScene.js]: Use requestAnimationFrame instead

File: src/modules/UI/selectors.js [hunk 24]
```diff
@@ -135,6 +149,10 @@ export const getSceneState = (state: State, sceneKey: string) => {
 }
 
 export const getExchangeRate = (state: State, fromCurrencyCode: string, toCurrencyCode: string): number => {
+  if (fromCurrencyCode === FIO_STR || toCurrencyCode === FIO_STR) {
+    return 1
```
Comment (paullinator) [inline src/modules/UI/selectors.js]: Put a comment to fix this. ie 

`// TODO: add real exchange rate`

File: src/components/materialWrappers/Select.js (hunk 9 above)
Comment (ericbutz) [reply src/components/materialWrappers/Select.js]: 1/10 discussion - Leave in for now, Paul will research.

File: src/actions/WalletOptionsActions.js (hunk 1 above)
Comment (andreyvEze) [reply src/actions/WalletOptionsActions.js]: ok

File: src/components/scenes/FioAddressConfirmScene.js (hunk 2 above)
Comment (andreyvEze) [reply src/components/scenes/FioAddressConfirmScene.js]: will be removed

File: src/connectors/scenes/FioAddressDetailsConnector.js (hunk 21 above)
Comment (andreyvEze) [reply src/connectors/scenes/FioAddressDetailsConnector.js]: ok

File: src/components/scenes/FioAddressConfirmScene.js (hunk 3 above)
Comment (andreyvEze) [reply src/components/scenes/FioAddressConfirmScene.js]: will be removed

File: src/locales/en_US.js (hunk 4 above)
Comment (andreyvEze) [reply src/locales/en_US.js]: will be removed

File: src/components/scenes/FioAddressRegisterScene.js [hunk 25]
```diff
@@ -0,0 +1,200 @@
...
+  fioWallets: EdgeCurrencyWallet[]
+}
+
+export type DispatchProps = {
+  changeFioAddressName(fioAddressName: string): any,
+  createCurrencyWallet: (walletName: string, walletType: string) => any
+}
+
+type Props = StateProps & DispatchProps & GoToSceneProps & CheckConnectivityProps
+
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/components/scenes/FioAddressRegisterScene.js]: will be refactored

File: src/actions/SettingsActions.js [hunk 26]
```diff
@@ -358,3 +359,22 @@ export const setDeveloperModeOn = (developerModeOn: boolean) => (dispatch: Dispa
...
+})
+
+export const goToScene = (route: string, params?: Object) => (dispatch: Dispatch, getState: GetState) => {
+  const state = getState()
+  const isConnected = getNetworkConnectivity(state)
+  if (!isConnected) return networkAlert()
+  Actions[route](params || {})
+}
+
+export const checkConnectivity = () => (dispatch: Dispatch, getState: GetState) => {
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/actions/SettingsActions.js]: will be refactored

File: src/components/Main.ui.js [hunk 27]
```diff
@@ -165,10 +177,18 @@ export default class Main extends Component<Props> {
         if (url) {
           this.doDeepLink(url)
         }
-        // this.navigate(url);
+        // this.navigate(url)
       })
       .catch(showError)
     Linking.addEventListener('url', this.handleOpenURL)
+
+    this.netInfoListener = NetInfo.addEventListener(state => {
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/components/Main.ui.js]: ok

File: src/actions/DeleteFioWalletModalActions.js [hunk 28]
```diff
@@ -0,0 +1,43 @@
...
+
+import { launchModal } from '../components/common/ModalProvider.js'
+import { DELETE } from '../constants/indexConstants.js'
+import s from '../locales/strings.js'
+import { getAccount, getWalletName } from '../modules/Core/selectors.js'
+import Text from '../modules/UI/components/FormattedText/index'
+import OptionIcon from '../modules/UI/components/OptionIcon/OptionIcon.ui'
+import type { Dispatch, GetState } from '../types/reduxTypes'
+
+export const showDeleteFioWalletModal = (walletId: string) => async (dispatch: Dispatch, getState: GetState) => {
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/actions/DeleteFioWalletModalActions.js]: ok

File: src/styles/scenes/FioAddressConfirmStyle.js [hunk 29]
```diff
@@ -0,0 +1,101 @@
...
+    justifyContent: 'flex-start',
+    alignItems: 'center'
+  },
+  underlay: {
+    color: `${THEME.COLORS.PRIMARY}${THEME.ALPHA.LOW}`
+  },
+  errMsg: {
+    marginTop: scale(20),
+    color: THEME.COLORS.ACCENT_RED,
+    fontSize: 14,
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/styles/scenes/FioAddressConfirmStyle.js]: will be added

File: src/util/utils.js (hunk 5 above)
Comment (andreyvEze) [reply src/util/utils.js]: ok

File: src/util/utils.js (hunk 10 above)
Comment (andreyvEze) [reply src/util/utils.js]: we need to show 2 numbers after the dot or all as needed

File: src/util/utils.js (hunk 6 above)
Comment (andreyvEze) [reply src/util/utils.js]: ok

File: src/components/scenes/FioAddressListScene.js (hunk 13 above)
Comment (andreyvEze) [reply src/components/scenes/FioAddressListScene.js]: ok

File: src/components/scenes/FioAddressRegisterScene.js (hunk 14 above)
Comment (andreyvEze) [reply src/components/scenes/FioAddressRegisterScene.js]: ok

File: src/components/scenes/FioAddressRegisterScene.js (hunk 15 above)
Comment (andreyvEze) [reply src/components/scenes/FioAddressRegisterScene.js]: ok

File: src/modules/FioAddress/action.js (hunk 16 above)
Comment (andreyvEze) [reply src/modules/FioAddress/action.js]: will be refactored

File: src/components/scenes/FioAddressRegisterScene.js (hunk 17 above)
Comment (andreyvEze) [reply src/components/scenes/FioAddressRegisterScene.js]: ok

File: src/actions/FioActions.js (hunk 18 above)
Comment (andreyvEze) [reply src/actions/FioActions.js]: will be removed

File: src/modules/UI/selectors.js (hunk 20 above)
Comment (andreyvEze) [reply src/modules/UI/selectors.js]: will be refactored

File: src/modules/UI/selectors.js (hunk 24 above)
Comment (andreyvEze) [reply src/modules/UI/selectors.js]: ok

File: src/actions/SettingsActions.js [hunk 30]
```diff
@@ -358,3 +359,22 @@ export const setDeveloperModeOn = (developerModeOn: boolean) => (dispatch: Dispa
     })
     .catch(showError)
 }
+
+export const changeConnectivity = (isConnected: boolean) => ({
+  type: 'UI/SETTINGS/NETWORK_STATUS',
+  data: { isConnected }
+})
+
+export const goToScene = (route: string, params?: Object) => (dispatch: Dispatch, getState: GetState) => {
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/actions/SettingsActions.js]: will be refactored

File: src/components/Main.ui.js (hunk 12 above)
Comment (andreyvEze) [reply src/components/Main.ui.js]: will be removed and refactored

File: src/util/utils.js (hunk 11 above)
Comment (andreyvEze) [reply src/util/utils.js]: Refactored

File: yarn.lock (hunk 8 above)
Comment (andreyvEze) [reply yarn.lock]: required by react-native-material-dropdown

File: yarn.lock (hunk 7 above)
Comment (andreyvEze) [reply yarn.lock]: required by react-native-material-dropdown

File: src/connectors/scenes/FioAddressConfirmConnector.js (hunk 19 above)
Comment (andreyvEze) [reply src/connectors/scenes/FioAddressConfirmConnector.js]: get default value for currency

File: src/components/scenes/FioAddressRegisterScene.js [hunk 31]
```diff
@@ -0,0 +1,210 @@
...
+    const isValid = !(fioAddressLength === 0 || fioAddressLength > 20 - 5 || !regexValidation)
+
+    if (this.bcValidationTimer) {
+      clearTimeout(this.bcValidationTimer)
+      this.bcValidationTimer = null
+    }
+    if (!checkConnectivity(this.props.isConnected, s.strings.fio_network_alert_text)) return
+    if (isValid) {
+      this.bcValidationTimer = setTimeout(async () => {
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/components/scenes/FioAddressRegisterScene.js]: changed

File: src/components/scenes/FioAddressRegisterScene.js [hunk 32]
```diff
@@ -0,0 +1,210 @@
...
+
+  handleFioAddressChange = async (fioAddressChanged: string) => {
+    if (!this.wallet) {
+      return this.setState({
+        isAvailable: null
+      })
+    }
+    fioAddressChanged = fioAddressChanged.replace(this.state.replaceRegex, '')
+    const fioAddressLength = fioAddressChanged.length
+    const regexValidation = this.wallet ? this.wallet.otherMethods.validateFioAddress(fioAddressChanged + FIO_DOMAIN_DEFAULT) : false
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/components/scenes/FioAddressRegisterScene.js]: done

File: src/components/common/FormField.js [hunk 33]
```diff
@@ -44,7 +44,8 @@ class FormField extends Component {
     forceFocus: false,
     returnKeyType: 'go',
     label: '',
-    keyboardType: 'default'
+    keyboardType: 'default',
+    multiline: true
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/components/common/FormField.js]: change to default 'false'

File: src/connectors/scenes/FioAddressDetailsConnector.js [hunk 34]
```diff
@@ -0,0 +1,23 @@
...
+
+  const out: StateProps = {
+    fioAddressName: fioAddress.fioAddressName,
+    expiration: fioAddress.expiration,
+    fee_collected: fioAddress.fee_collected
+  }
+  return out
+}
+
+export default connect(
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/connectors/scenes/FioAddressDetailsConnector.js]: done

File: src/connectors/scenes/FioAddressConfirmConnector.js [hunk 35]
```diff
@@ -0,0 +1,38 @@
...
+  changeConfirmSelectedWallet: (selectedWallet: EdgeCurrencyWallet | null, expiration: Date, fee_collected: number) =>
+    dispatch({
+      type: 'FIO/FIO_ADDRESS_UPDATE_SELECTED_WALLET',
+      data: { selectedWallet, expiration, fee_collected }
+    })
+})
+
+export default connect(
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/connectors/scenes/FioAddressConfirmConnector.js]: done

File: src/connectors/scenes/FioAddressConnector.js [hunk 36]
```diff
@@ -0,0 +1,10 @@
+// @flow
+
+import { connect } from 'react-redux'
+
+import { FioAddressScene } from '../../components/scenes/FioAddressScene'
+
+export default connect(
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/connectors/scenes/FioAddressConnector.js]: done

File: src/connectors/scenes/FioAddressListConnector.js [hunk 37]
```diff
@@ -0,0 +1,31 @@
...
+
+const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
+  setFioAddress: (fioAddressName: string, expiration: string) =>
+    dispatch({
+      type: 'FIO/FIO_ADDRESS_SET_FIO_ADDRESS',
+      data: { fioAddressName, expiration }
+    })
+})
+
+export default connect(
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/connectors/scenes/FioAddressListConnector.js]: done

File: src/connectors/scenes/FioAddressRegisterConnector.js [hunk 38]
```diff
@@ -0,0 +1,33 @@
...
+const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
+  changeFioAddressName: (fioAddressName: string) =>
+    dispatch({
+      type: 'FIO/FIO_ADDRESS_UPDATE_FIO_ADDRESS_NAME',
+      data: { fioAddressName }
+    }),
+  createCurrencyWallet: (walletName: string, walletType: string) => dispatch(createCurrencyWallet(walletName, walletType, 'iso:USD', false, false))
+})
+
+export default connect(
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/connectors/scenes/FioAddressRegisterConnector.js]: done

File: src/reducers/scenes/SettingsReducer.js [hunk 39]
```diff
@@ -105,7 +106,8 @@ export type SettingsState = {
     '2000': boolean,
     '20000': boolean,
     '200000': boolean
-  }
+  },
+  isConnected: boolean
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/reducers/scenes/SettingsReducer.js]: moved

File: src/util/utils.js [hunk 40]
```diff
@@ -628,3 +630,22 @@ export const getYesterdayDateRoundDownHour = () => {
   const yesterday = date.setDate(date.getDate() - 1)
   return new Date(yesterday).toISOString()
 }
+
+export const checkConnectivity = (isConnected: boolean, alertText: string = ''): boolean => {
```
[the remark this answers is not in the document]
Comment (andreyvEze) [reply src/util/utils.js]: refactored

**Pull request merged**